io/plugins/eu.esdihumboldt.hale.io.csv/src/eu/esdihumboldt/hale/io/csv/writer/internal/CSVInstanceWriter.java

```diff
@@ -76,27 +79,64 @@ public boolean isCancelable() {
 	protected IOReport execute(ProgressIndicator progress, IOReporter reporter)
 			throws IOProviderConfigurationException, IOException {
 
-		boolean solveNestedProperties = getParameter(
-				InstanceTableIOConstants.SOLVE_NESTED_PROPERTIES).as(Boolean.class, false);
-		// XXX what does "solve nested properties" mean?
+		URI location = getTarget().getLocation();
+
+		String filenameWithType = location.getPath();
+		String filePath = Paths.get(location).getParent().toString();
```

Comment:
An URI is not necessarily a file URI. Also, there is the possibility that the location is null, for cases where only an output stream is provided.
There are some writers that can only write to files (e.g. Geopackage), but the CSV writer so far could also handle other targets (see also comment further below).